conjure/servicewriter.go

```diff
@@ -1018,25 +1049,20 @@ func returnTypesForEndpoint(endpointDefinition spec.EndpointDefinition, info typ
 	var returnTypes []expression.Type
 	imports := make(StringSet)
 	if endpointDefinition.Returns != nil {
-		var goType string
-		returnBinary, err := isReturnTypeSpecificType(endpointDefinition.Returns, visitors.IsBinary)
+		typer, err := visitors.NewConjureTypeProviderTyper(*endpointDefinition.Returns, info)
 		if err != nil {
 			return nil, nil, err
 		}
+		returnBinary, err := isReturnTypeSpecificType(endpointDefinition.Returns, visitors.IsBinary)
 		if returnBinary {
 			// special case: "binary" type resolves to []byte in structs, but indicates a streaming response when
 			// specified as the return type of a service, so use "io.ReadCloser".
-			goType = types.IOReadCloserType.GoType(info)
+			typer = types.MapBinaryTypeToReadCloserType(typer)
 			imports.AddAll(NewStringSet(types.IOReadCloserType.ImportPaths()...))
```

Comment:
Per feedback on the `MapBinaryTypeToReadCloserType` function, the `IOReadCloserType` import should probably be part of that logic (seems strange to require caller to have to know to add this)